src/Extensions/ElementalAreaExtension.php

```diff
@@ -2,71 +2,93 @@
+            $logger->error($exception->getMessage(), [
+                'exception' => $exception,
+            ]);
+            $this->controllers = new ArrayList();
+        }
+    }
+
+    /**
+     * @return false|ArrayList
+     * @throws \Exception
+     */
+    public function ElementControllersWithRows()
+    {
+        if (!$this->controllers->count()) {
             return false;
         }
 
         //Check for first row
-        $first = $controllers->first();
-        if($first->ClassName == ElementRow::class){
+        $first = $this->controllers->first();
+        if ($first->ClassName == ElementRow::class) {
             $first->setIsFirstRow(true);
-        }else{
+        } else {
             $createdFirstElement = ElementRow::create();
             $createdFirst = $createdFirstElement->getController();
             $createdFirst->setIsFirstRow(true);
-            $controllers->unshift($createdFirst);
+            $this->controllers->unshift($createdFirst);
         }
 
-        //Check for last row
-        $last = $controllers->last();
-        if($last->ClassName == ElementRow::class){
+        // Check for last row
+        $last = $this->controllers->last();
+
+        if ($last->ClassName == ElementRow::class) {
             $last->setIsLastRow(true);
-        }else{
+        } else {
             $createdLastElement = ElementRow::create();
             $createdLast = $createdLastElement->getController();
             $createdLast->setIsLastRow(true);
-            $controllers->push($createdLast);
+            $this->controllers->push($createdLast);
         }
 
         $previousRow = false;
-        $previousKey = false;
-        $previousController = false;
-        foreach($controllers as $key => $controller){
-            if($controller->ClassName == ElementRow::class){
-                if($previousRow){
-                    $controller->setPreviousRow($previousRow->getElement());
-                }
-                if($previousController && $previousController instanceof ElementFullWidthController){
-                   $controller->setIsFirstRow(true);
-                }
-                $previousRow = $controller;
-            }else if($controller instanceof ElementFullWidthController){
-                if($previousController && $previousController->ClassName == ElementRow::class){
-                    unset($controllers[$previousKey]);
-                }
-                if($key > 1){
-                    $controller->setNeedClosing(true);
-                }
-                $next = array_key_exists($key + 1, $controllers->toArray()) ? $controllers->toArray()[$key + 1] : false;
-                if($next && $next->ClassName != ElementRow::class){
-                    $controller->setAfter(ElementRow::create());
+        foreach ($this->controllers as $key => $controller) {
+            if ($controller) {
+                if ($controller->ClassName == ElementRow::class) {
```

Comment:
These if statements can be combined, and ideally even reversed with a `continue`

so:

```php
if (!$controller || $controller->ClassName !== ElementRow::class) {
    continue;
}
```